src/yb/yql/pggate/pg_doc_op.cc

```diff
@@ -346,7 +359,13 @@ Status PgDocOp::SendRequestImpl(ForceNonBufferable force_non_bufferable) {
   // the request. We update instrumentation for writes sexactly once, after successfully sending an
   // RPC request to the underlying storage layer.
   if (IsWrite()) {
-    pg_session_->metrics().WriteRequest(ResolveRelationType(*pgsql_ops_.front(), table_));
+    auto relation_type = ResolveRelationType(*pgsql_ops_.front(), table_);
```

Comment:
When/where do these get unset?

I only see
`UnsetWaitEventInfo();`

run `if (!IsWrite()) {`

Reply:
Is it in 
`in_flight_ops_.front().future.session()->UnsetWaitEventInfo();`

Reply:
Yeah, that's the one. Ideally we should have the `SetWaitEventInfo` there as well, because that's where the actual waiting is, but I didn't have enough time to figure out how to distinguish between catalog and storage requests there, so I set the wait event here.